### PR TITLE
[Dubbo-7483][3.0] Fix stringUrls cache was deleted when startup

### LINKED ISSUE #7483: [3.0] stringUrls cache was deleted when startup
https://github.com/apache/dubbo/pull/7476/files#diff-b00639ea715ae8698865e33943cacc6a08ba0985e78980b8b74c085e4b12188bR165-R168
This pull request wants to delete stringUrls cache when the provider is empty or unsubscribe is called. but forgot to check path `/providers` in the first case, cause stringUrls was cleared every time when providers change. And this pull request also forgot to clear stringAddress and stringParam cache.
image

### PR DESCRIPTION
**Pull request by kevinw66**
## What is the purpose of the change

Fix #7483

### CODE REVIEW

**Review by chickenlj — APPROVED** [review]
LGTM.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/CacheableFailbackRegistry.java [hunk 1]
```diff
@@ -138,42 +136,28 @@ protected static int getIntConfig(String key, int def) {
...
         return new ArrayList<>(newURLs.values());
     }
 
     protected List<URL> toUrlsWithEmpty(URL consumer, String path, Collection<String> providers) {
-        List<URL> urls;
-        if (CollectionUtils.isEmpty(providers)) {
-            urls = new ArrayList<>(1);
-            // clear cache on empty notification: unsubscribe or provider offline
-            stringUrls.remove(consumer);
+        List<URL> urls = new ArrayList<>(1);
```
Comment (kimmking) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/CacheableFailbackRegistry.java:146]: plz resolve the conflicts first

**Review by kimmking — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/CacheableFailbackRegistry.java (hunk 1 above)
Comment (kevinw66) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/CacheableFailbackRegistry.java:172]: done

**Review by AlbumenJ — APPROVED** [review]
LGTM.

**Pull request merged**